src/main/java/filter/lexer/Lexer.java

```diff
@@ -26,7 +26,7 @@
 
             new Rule("\\(", TokenType.LBRACKET),
             new Rule("\\)", TokenType.RBRACKET),
-            new Rule("\\\"", TokenType.QUOTE),
+            new Rule("\\\"([^(\\\")]+)\\\"", TokenType.QUOTED_CONTENT),
```

Comment:
The parentheses around the `[...]+` are unnecessary I think, because this simple lexer doesn't use capturing groups.

Are the parentheses inside necessary? Did you intend to disallow `label:"(abc)"`?